### DISCUSSION TITLE
Breaking change with custom docker image prefix

### DISCUSSION THREAD

**Discussion by bobvandevijver** [renovatebot/renovate#18760] [Suggest an Idea]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.17.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.5.2

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

PR #17656 introduced a breaking change by renaming the docker image used for the PHP sidecar container. Users that set `RENOVATE_DOCKER_IMAGE_PREFIX` because they need to use a container with additional tooling installed will now experience Renovate being broken for them as it tries to pull an non-existing container.

This is probably also relevant for #​16830 & #16828.

We will probably work around this by renaming our container from `renovate-crm/php` to `renovate-crm-php/sidecar`. This is possible as this particular project only uses renovate for php, but will come back when other managers are added.

I'm posting this as a bug report as I believe this is an unexpected side-effect of the linked change.

It would be nice to be able to override the used image name per manager, maybe even the full image location so only the one image that needs to be custom can be configured.

### Relevant debug logs

<details><summary>Logs</summary>

```
WARN: artifactErrors (repository=backoffice/crm, branch=renovate/drenso-symfony-shared-2.x)
       "artifactErrors": [
         {
           "lockFile": "composer.lock",
           "stderr": "Command failed: docker pull docker-repo.something/renovate-crm/sidecar\nError response from daemon: manifest for docker-repo.something/renovate-crm/sidecar:latest not found: manifest unknown: manifest unknown\n"
         }
       ]
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
We are moving all package managers to use the `sidecar` image, more than half are done. I suggest if you have any other images with custom installs then add the tooling to `sidecar` in advance now. e.g. `go`, `ruby` and `nuget`. You can follow along at https://github.com/renovatebot/renovate/issues/13683

I apologize that this change broke you, but we won't roll this change back, so I'm not sure there's any reason to keep this issue open.

Assuming that we consolidate all images onto `sidecar`, will you have any reason to need to be able to configure the image name to be used?

**Comment by viceice** [comment]
I think we should convert this to a discussion, as this is expected change

**Comment by bobvandevijver** [comment]
@rarkins This is my current process:

1. Start with renovate/php
1. Install extra tooling required by this project (java in this case)
1. Install extra extensions that require php to be installed

If I extend the image from sidecar, it would require an extra install php step, which will be rerun by renovate when started, right? It would also include the extra tooling to be available when other managers are run, which do not need this tooling installed.

Also, if I understand correctly, every time a branch is run, the tooling is reinstalled when using the sidecar image right? That feels quite inefficient...

**Comment by viceice** [comment]
@bobvandevijver  the sidecar has only the bare minimum deps installed, no tools by default. the `install-tool` will also skip install, if that version is already installed.

The sidecar has also a download cache, so installing a tool will be a simple and fast extract or copy of some files, which is mostly done in less than a second

**Comment by bobvandevijver** [comment]
I was literally on the verge of closing the issue, but okay.

I think I have found a method which wil work for what I need. For reference:

- Start with `renovate/sidecar`
- Install Java dependency
- Install desired PHP version
- Install needed PHP extensions
- `export IGNORED_TOOLS=php` <- this is important, as the extensions above are depending on the PHP version installed earlier
